### ISSUE TITLE
Start of second server causes basic_string::_S_construct null not valid error

### ISSUE THREAD

**Issue by grahamreeds** [actor-framework/actor-framework#565]
Start with simple_http_broker example.

Add a second server on differing port to the first and the second server crashes the application with a std::logic_error on the spawn_server():

> *** listening on port 5061
terminate called after throwing an instance of 'std::logic_error'
  what():  basic_string::_S_construct null not valid
Aborted (core dumped)
> 

Fails in topic/udp, and develop. 

Master fails with a different error: 
> *** listening on port 5061
> *** cannot spawn tcp server: system_error(network_syscall_failed, ("bind", "Address already in use"))

Which seems to incorrectly try and reuse the same port.

**Comment by josephnoir** [comment]
What do you mean by adding a second server? Using this code and the default port (0 in the config), both servers are started of different ports (tested on macOS and my Ubuntu 17.04 VM). I can produce a segfault though, when I start both servers on the same port.

Here is my interpretation of adding a second server:
```
void caf_main(actor_system& system, const config& cfg) {
  auto server_actor = system.middleman().spawn_server(server, cfg.port);
  if (!server_actor) {
    cerr << "*** cannot spawn server: "
         << system.render(server_actor.error()) << endl;
    return;
  }
  auto server_actor2 = system.middleman().spawn_server(server, cfg.port);
  if (!server_actor2) {
    cerr << "*** cannot spawn server: "
         << system.render(server_actor2.error()) << endl;
    return;
  }
  cout << "*** listening on port " << cfg.port << endl;
  cout << "*** to quit the program, simply press <enter>" << endl;
  // wait for any input
  std::string dummy;
  std::getline(std::cin, dummy);
  // kill server
  anon_send_exit(*server_actor, exit_reason::user_shutdown);
}
```

**Comment by grahamreeds** [comment]
Switched back to /develop

It fails with a non-zero port number.  If the port is zero then both servers start.  If you change the value (to 5061 for instance) its fails.

**Comment by josephnoir** [comment]
Ok, so the expected behavior would be
* both servers to run on different ports if the port argument is 0,
* only the first server is started successfully if the port arguments are not zero and the same,
* both servers run on their respective ports if the arguments are different and valid port.

This [commit](https://github.com/actor-framework/actor-framework/pull/566) should fix that. I created a PR in case there is more I am missing here.

Oh, yeah, I only tested that on develop.
